src/modules/plus/filter_dynamicgpstext.c

```diff
@@ -0,0 +1,1714 @@
+}
+
+/** Checks if file is processed or needs to be read from disk,
+ *  finds the closest (time-wise) gps_point and uses it to call gps_point_to_output
+*/
+static void get_next_gps_frame_data (mlt_filter filter, mlt_frame frame, const time_t video_time_synced, const time_t video_time_original, const char* keyword, char gps_text[MAX_TEXT_LEN])
+{
+	private_data* pdata = (private_data*)filter->child;
+	gps_point* gps_points = pdata->gps_points;
+
+	mlt_properties properties = MLT_FILTER_PROPERTIES( filter );
+	char* location_filename = mlt_properties_get( properties, "gps.file");
+	int crt_smooth = mlt_properties_get_int(properties, "smoothing_value");
+	
+	//if there's no file selected just return
+	if ( !location_filename || !strcmp(location_filename, "") ) {
+		//mlt_log_warning(NULL, "filter_dynamicgps.c get_next_gps_frame_data gps_file string null, %s", location_filename==0?"null":location_filename);
+		return;
+	}
+	
+	//check to see if we need to trim the gps start time:
+	char* gps_processing_start_time = mlt_properties_get(properties, "gps_processing_start_time");
+	if (gps_processing_start_time != NULL) {
+		time_t gps_proc_t = 0;
+		if (strlen(gps_processing_start_time)!=0 && strcmp(gps_processing_start_time, "yyyy-MM-dd hh:mm:ss"))
+			gps_proc_t = datetimeXMLstring_to_seconds(gps_processing_start_time, "%Y-%m-%d %H:%M:%S");
+		//if different start time is selected we must clear extra data, resmooth with level and recalculate
+		if (gps_proc_t != pdata->gps_proc_start_t) {
+			//mlt_log_info(NULL, "filter_dynamicgps.c get_next_gps_frame_data gps_processing_start_time=%s -> %d", gps_processing_start_time, gps_proc_t);
+			pdata->gps_proc_start_t = gps_proc_t;
+			process_gps_smoothing(filter, 0);
+			process_gps_smoothing(filter, crt_smooth);
+			recalculate_gps_data(filter, crt_smooth);
+		}
+	}
+
+	//check if the gps file has been changed, if not, just use current data
+	if (strcmp(pdata->last_filename, location_filename))
+	{
+		//mlt_log_info(NULL, "filter_dynamicgps.c get_next_gps_frame_data: last_filename (%s) != entered_filename (%s), reading data from file", pdata->last_filename, location_filename);
+		cleanup_priv_data(pdata);
+		strcpy(pdata->last_filename, location_filename);
+		process_gps_file(filter, frame, location_filename);
+		gps_points = pdata->gps_points;
+	}
+	
+	//process smoothing
+	if (pdata->last_smooth_lvl != crt_smooth) {
+		pdata->last_smooth_lvl = crt_smooth;
+		process_gps_smoothing(filter, crt_smooth);
+	}
+
+	int i_gps = -1;
+	if ((i_gps = binary_search_gps(filter, video_time_synced)) == -1)
+		return;
+
+	//write the requested info in gps_text
+	gps_point_to_output(gps_points[i_gps], keyword, video_time_original, gps_text);
+}
+
+/** Returns the unix time (seconds) of "Media Created" metadata, or file's "Modified Time" from OS
+ */
+static time_t get_original_file_time_seconds (mlt_filter filter, mlt_frame frame)
```

Comment:
Consider seeing if this functions works for you:
https://github.com/mltframework/mlt/blob/master/src/framework/mlt_producer.c#L1201

Reply:
Yes, will switch to it as it looks more robust.

Do you know if there's a function that will return timecode in seconds (this: mlt_properties_frames_to_time() returns a string that I have to parse).

Reply:
There is not already a function that returns seconds.

Upon closer inspection, I see that you are already using mlt_producer_get_creation_time() - but then you fall back to checking the file_info yourself. It is my intention that mlt_product_get_creation_time() will make its best effort to provide a time. If that is not working for you, let's fix that function. If it is working for you, you can remove the "method 2" below.

Regarding mlt_properties_frames_to_time(), it may be better for you to use mlt_producer_position() or mlt_properties_get_position() and then divide by the profile frame rate. That would avoid string parsing.

Reply:
I didn't notice at first that it fallbacks to mtime. I'll update.